src/main/java/org/gridsuite/network/map/NetworkMapService.java

```diff
@@ -251,28 +251,42 @@ private static ThreeWindingsTransformerMapData toMapData(ThreeWindingsTransforme
         return builder.build();
     }
 
-    public List<SubstationMapData> getSubstations(UUID networkUuid) {
+    public List<SubstationMapData> getSubstations(UUID networkUuid, List<String> substationsId) {
         Network network = getNetwork(networkUuid);
-        return network.getSubstationStream().map(NetworkMapService::toMapData).collect(Collectors.toList());
+        return network.getSubstationStream()
```

Comment:
should we implement this filter in the database ?

Reply:
Code has been reimplemented